syntax = "proto2";

//---------------------------------

enum ProtocolID
{
   PROT_ID_NONE 	= 0;
   PROT_ID_DNP3		= 1;
}

enum MessageType
{
  MSG_TYPE_NONE = 0;
  MSG_TYPE_GW = 1;
  MSG_TYPE_USER = 2;
}
//---------------------------------

message HeaderMessage  // Header message, will be a required field
{
  optional ProtocolID protocol		= 1; // protocol ID
  optional MessageType msgType		= 2; // message type
  optional uint64 serial		= 3; // device serial
  required uint64 ticket		= 4; // ticket number
  required uint64 gwid			= 5; // gateway id  
}

//---------------------------------

